include/saltatlas/dnnd/detail/dnnd_kernel.hpp

```diff
@@ -352,8 +357,7 @@ class dnnd_kernel {
       m_comm.cout0() << "Filling initial index took (s)\t"
                      << init_timer.elapsed() << std::endl;
       m_comm.cout0() << "#of generated initial neighbors: "
-                     << ygm::sum(m_knn_heap_table.size() * init_k, m_comm)
-                     << std::endl;
+                     << ygm::sum(num_random_neighbors) << std::endl;
```

Comment:
Sorry, there was a mistake in my proposal. `m_comm` was missing there.

It must be something like:
```c++
ygm::sum(num_random_neighbors, m_comm)
```